### ISSUE TITLE
The accuracy on the cityscape

### ISSUE THREAD

**Issue by FengLoveBella** [zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow#9]
Could you tell me the accuracy on the cityscape via using your code. I want to check it is my fault or just the performance of the deeplabv2. I got a very low Mean IoU on cityscape. Thank you very much. @zhengyang-wang

**Comment by zhengyang-wang** [comment]
It is harder to reproduce the deeplab results on cityscape than on PASCAL.
The performance depends on the training patch size (typically larger is better but we have limited memory) and the way you do testing. Note that in deeplab paper, they claimed that they performed inference on small overlapping patches, while this code tests on whole image. Also, the training set of cityscape has two parts (fine and coarse labels). All these things will affect the final results.
Could you share your setting and results, please? We can discuss and see whether the results are reasonable.

**Comment by FengLoveBella** [comment]
I got a 0.656 IOU, but I think it is a little lower. 
I use the resnet50 provided in your code, but I compare it with the standard resnet50, I found somewhere is different, such as stride in the '_dilated_bottle_resblock'.  You set it almost to 1, but the standard stride is 2 in my opinion, which one is correct, or I miss something? @zhengyang-wang

**Comment by zhengyang-wang** [comment]
The paper used resnet101-based models. It may improve results compared to resnet50.
Could you specify what standard you were using to do the comparison and which stride (there are several convolution operations in a bottleneck block)? For blocks with dilated convolutions, the stride is always 1. For a regular bottleneck block, the stride is sometimes 2 for some convolution operations, depending on whether the spatial size changes in this block. If my stride setting is wrong, the spatial size will be wrong. Thanks.

**Comment by FengLoveBella** [comment]
Alright, I see. And another question, in your code, how many batch size is small, and if the batch_size in my code is 4, do you think, I can set is_training to true in _batch_norm.

**Comment by FengLoveBella** [comment]
And about the resnet50, I just download a caffe prototxt, I compare your code with it, and I found it is a little different in the stride,  in the block which kernel size is 1024 and 2048.

**Comment by FengLoveBella** [comment]
And, to be honest,  I am a little confused between is_training and training.  If I set trainable=True, and is training_false, what is that mean. And set trainable=False, is_training=true/false, what is that mean? Tks.

**Comment by zhengyang-wang** [comment]
1. It is hard to say what batch size is small. But if you take a look at successful image classification models, the batch size is usually larger than 100. We are doing segmentation (dense prediction) so that we cannot have such a large batch size due to memory limitation.
2. Original ResNet50 even does not have dilated convolutions. Our task is segmentation instead of classification. You were talking about different things.
3. From the documentation of tensorflow:
is_training: Whether or not the layer is in training mode. In training mode it would accumulate the statistics of the moments into moving_mean and moving_variance using an exponential moving average with the given decay. When it is not in training mode then it would use the values of the moving_mean and the moving_variance.
trainable: If True also add variables to the graph collection GraphKeys.TRAINABLE_VARIABLES (see tf.Variable).

**Comment by FengLoveBella** [comment]
Tks I see
